api/services/CasperServices.js

```diff
@@ -113,11 +117,12 @@ class CasperServices {
 					const { execution_results, deploy } = result;
 					return {
 						hash: deploy.hash,
-						status: !execution_results.length
-							? 'pending'
-							: execution_results.some((rs) => rs.result.Failure)
-							? 'fail'
-							: 'success',
+						status:
+							!execution_results || !execution_results.length
```

Comment:
Should we extract this complex logic to a local function?

Reply:
I think no need, the logic is matching with function name